### PR TITLE
Custom callbacks allowed

### PR DESCRIPTION
**Pull request by indrimuska**
Perform customization of callbacks onSelect / onChangeMonthYear / beforeShow without loosing mobile style.

Demo: http://jsbin.com/indrimuska/42/edit.

### CODE REVIEW

File: jquery.mobile.datepicker.js [hunk 1]
```diff
@@ -70,11 +42,25 @@
...
         var calendar, interval,
           that = this;
+        
+        $.each([ 'onSelect', 'onChangeMonthYear', 'beforeShow' ], function(key, val){
+			that.options[ '_'+val ] = that.options[ val ];
+			that.options[ val ] = function(){
+				var args = arguments;
+				setTimeout(function(){
+					that.addMobileStyle();
+					if (callback = that.options[ '_'+val ]) {
```
Comment (arschmitz) [inline jquery.mobile.datepicker.js]: you are assigning a global variable in an if statement here i think what you want is to just do `if( that.options[ "_" +  val ]) {` then below do `that.options[ "_" +  val ].apply( null, args );`

**Comment by indrimuska** [comment]
Yes, it's the same.
There are many ways to do that:
1. Assign in the if-statement (my way):
   `if ( callback = that.options[ '_'+val ] ) {
   callback.apply( null, args );
   }`
2. Without assignment (your way):
   `if ( that.options[ '_'+val ] ) {
   that.options[ '_'+val ].apply( null, args );
   }`
3. Assignment before the if-statement:
   `var callback = that.options[ '_'+val ];
   if ( callback ) {
     callback.apply( null, args );
   }`

My goal is to rename user callback, i.e. `onSelect`, into `_onSelect`, and then recall the callback after to add the mobile style to the datepicker. In this way, ui-date triggers the overridden `onSelect` which takes care to execute `addMobileStyle();` and `_onSelect` if set.

**Comment by arschmitz** [comment]
@indrimuska yes what your trying to do is correct but you are creating a global variable because you are not using `var` and we cannot create global variables. I would like the non assignment way because assignment offers no benefit here

**Comment by indrimuska** [comment]
Thank you @arschmitz, I've followed your advice.

**Comment by arschmitz** [comment]
@indrimuska can you make a seperate pr for fixing the height of the days?

**Comment by indrimuska** [comment]
Ok, I'll do it asap.

**Pull request closed without merging**